shepherd_bms/src/segment.cpp

```diff
@@ -223,64 +223,68 @@ FaultStatus_t SegmentInterface::pullThermistors()
     // If polled too soon, just copy existing values from memory
 	if (!thermTimer.isTimerExpired())
 	{
-		for(uint8_t i=0; i<NUM_CHIPS; i++)
+		for(uint8_t c=0; c<NUM_CHIPS; c++)   
         {
-            memcpy(segmentData[i].thermistorReading, previousData[i].thermistorReading, sizeof(segmentData[i].thermistorReading));
+            memcpy(segmentData[c].thermistorReading, previousData[c].thermistorReading, sizeof(segmentData[c].thermistorReading));
         }
         return voltageError;
 	}
 
     uint16_t rawTempVoltages[NUM_CHIPS][6];
 
     // Rotate through all thermistor pairs (we can poll two at once)
-    for (int therm = 1; therm <= 16; therm++)
+    for (int therm = START_THERM; therm <= (END_THERM - 6); therm++) // therm range = 11 = ~12, two at a time = 6 iterations
 	{
+
+        
         // Sets multiplexors to select thermistors
         SelectTherm(therm);
-        SelectTherm(therm + 16);
+        SelectTherm(therm + 6);
```

Comment:
I don't think this will work. Pretty sure the one with (therm + 6) will for most cases overwrite what (therm) did

Reply:
i totally forgot about the first half vs second half thing we discussed, will fix this